pkg/sink/sink.go

```diff
@@ -150,6 +112,63 @@ func (r Sink) HandleEvent(response http.ResponseWriter, request *http.Request) {
 		r.EventListenerName, r.EventListenerNamespace, string(eventID), string(event), request.Header)
 }
 
+func (r Sink) processTrigger(t *triggersv1.EventListenerTrigger, request *http.Request, event []byte, eventID string, eventLog *zap.SugaredLogger) error {
+	if t == nil {
+		return errors.New("EventListenerTrigger not defined")
+	}
+	log := eventLog.With(zap.String(triggersv1.TriggerLabelKey, t.Name))
+
+	finalPayload, err := r.executeInterceptors(t, request, event, eventID, log)
+	if err != nil {
+		return err
+	}
+
+	rt, err := template.ResolveTrigger(*t,
+		r.TriggersClient.TektonV1alpha1().TriggerBindings(r.EventListenerNamespace).Get,
+		r.TriggersClient.TektonV1alpha1().TriggerTemplates(r.EventListenerNamespace).Get)
```

Comment:
Thanks for refactoring these functions out of the big one above!!

if we pass these clients into this funciton, then we can move the new functions into another package and have unit tests for them 🙏